src/secret-storage.ts

```diff
@@ -86,3 +91,48 @@ export interface PassphraseInfo {
     /** The number of bits to generate. Defaults to 256. */
     bits?: number;
 }
+
+/**
+ * Options for {@link SecretStorage#addKey}.
+ */
+export interface AddSecretStorageKeyOpts {
+    pubkey?: string;
+    passphrase?: PassphraseInfo;
+    name?: string;
+    key?: Uint8Array;
+}
+
+/**
+ * Return type for {@link SecretStorage#getKey}.
+ */
+export type SecretStorageKeyTuple = [keyId: string, keyInfo: SecretStorageKeyDescription];
+
+/**
+ * Return type for {@link SecretStorage#addKey}.
+ */
+export type SecretStorageKeyObject = { keyId: string; keyInfo: SecretStorageKeyDescription };
+
+/** Interface for managing account data on the server.
+ *
+ * A subset of {@link MatrixClient}.
+ */
+export interface AccountDataClient extends TypedEventEmitter<ClientEvent.AccountData, ClientEventHandlerMap> {
+    /**
+     * Get account data event of given type for the current user. This variant
+     * gets account data directly from the homeserver if the local store is not
+     * ready, which can be useful very early in startup before the initial sync.
+     *
+     * @param eventType - The type of account data
+     * @returns The contents of the given account data event.
+     */
+    getAccountDataFromServer: <T extends { [k: string]: any }>(eventType: string) => Promise<T>;
```

Comment:
I _think_ this could be written as `<T extends Record<string, any>>` if you were so inclined. I personally prefer the `Record` syntax for things like this and as far as I know they are completely equivalent in this case.

Doesn't block my approval anyhow.

Reply:
I agree with you, though in this case I was deliberately trying to move things without worrying too much about polishing their typescript definitions. (In this case, it's just been moved from https://github.com/matrix-org/matrix-js-sdk/pull/3244/files/e290ec3130a0206e1d329edf085a15662ae27cc5#diff-cef792a7fe5292ff218b44701c423e282ae1b8df338a537d576851781430da7fL46).

still, might as well fix this now you've pointed it out.

```suggestion
    getAccountDataFromServer: <T extends Record<string, any>>(eventType: string) => Promise<T>;
```